File: Task 3.vb
Module Module1

    Sub Main()
        Dim lostCharCount, lowerCharCount, upperCharCount, otherCharCount, numCount As Integer
        Dim position As String
        Dim inputChar As Char

        Console.Write("Input String: ")
        Dim input As String = Console.ReadLine()

        Console.Write("Which Character's position do you want to find?")
        Dim lostChar As Char = Console.ReadLine()

        For count = 1 To Len(input)
            If Mid(input, count, 1) = lostChar Then
                lostCharCount += 1
                position += Str(count)
            End If
        Next

        Console.WriteLine("")

        Console.WriteLine("The Character '" & lostChar & "' appears " & lostCharCount & " times at the Index/es: " & position)

        For count = 1 To Len(input)
            inputChar = Mid(input, count, 1)
            Select Case inputChar
                Case "a" To "z"
                    lowerCharCount += 1
                Case "A" To "Z"
                    upperCharCount += 1
                Case Else
                    otherCharCount += 1
            End Select
            If IsNumeric(inputChar) = True Then
                numCount += 1
            End If
        Next

        Console.WriteLine("Upper Case Character Count = " & upperCharCount)
        Console.WriteLine("Lower Case Character Count = " & lowerCharCount)
        Console.WriteLine("Numeric Character Count = " & numCount)
        Console.WriteLine("Other / Special Character Count = " & (otherCharCount - numCount))



        Console.ReadLine()
    End Sub

End Module

File: Python/Task_3.py
String = input("Enter String: ")
inputChar = input("Which character's position do you want to find?: ")
alphabets = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]

def find_char(userInput, lostChar):
    charCount = 0
    positions = ""
    count = 0
    for x in userInput:
        if x == lostChar:
            charCount += 1
            positions += str(count)
        count += 1
    return charCount, positions

def char_counter(userInput):
    upperCount = 0
    lowerCount = 0
    numCount = 0
    otherCount = 0
    for x in userInput:
        if x >= "A" and x <= "Z":
            upperCount += 1
        elif x >= "a" and x <= "z":
            lowerCount += 1
        elif x.isdigit() == True:
            numCount += 1
        else:
            otherCount += 1
    return upperCount, lowerCount, numCount, otherCount

Ucount, Lcount, Numcount, Othercount = char_counter(String)
charCount, positions = find_char(String, inputChar)

print(f"The character '{inputChar}' appears '{charCount}' times at the Index/es: {positions}")
print("")
print("Upper Case Character Count = ", Ucount)
print("Lower Case Character Count = ", Lcount )
print("Numeric Character Count = ", Numcount)
print("Other / Special Character Count = ", Othercount)
